Feature: Add dynamic RustDesk URL action button

ultimatejedi opened the pull request:
Adds a new button to the agent summary tab that is conditionally displayed if a URL Action named 'RustDesk_Connect' exists.

The button's URL is built dynamically by:
- Finding the URL Action 'pattern' from the Vuex store
- Looking up the agent's 'RustDesk_ID' custom field
- Replacing the {{agent.RustDesk_ID}} placeholder
- Leaving the {{global.*}} placeholder intact for the backend to process

NOTE: This requires a new Agent Custom Field labelled 'RustDesk_ID'
      There is also a modified version of the GetRustDeskID script that will populate the custom field.

wh1te909 committed:
init

wh1te909 committed:
Release 0.100.4

wh1te909 committed:
Release 0.100.5

wh1te909 committed:
Release 0.100.6

wh1te909 committed:
Release 0.100.7

wh1te909 committed:
Release 0.100.8

wh1te909 committed:
Release 0.100.9

wh1te909 committed:
Release 0.101.0

wh1te909 committed:
Release 0.101.3

wh1te909 committed:
Release 0.101.5

wh1te909 committed:
Release 0.101.7

wh1te909 committed:
Release 0.101.9

wh1te909 committed:
Release 0.101.11

wh1te909 committed:
Release 0.101.13

wh1te909 committed:
Release 0.101.16

wh1te909 committed:
Release 0.101.18

wh1te909 committed:
Release 0.101.20

wh1te909 committed:
Release 0.101.22

wh1te909 committed:
Release 0.101.25

wh1te909 committed:
Release 0.101.28

wh1te909 committed:
Release 0.101.29

wh1te909 committed:
Release 0.101.30

wh1te909 committed:
Release 0.101.31

wh1te909 committed:
Release 0.101.34

wh1te909 committed:
Release 0.101.35

wh1te909 committed:
Release 0.101.36

wh1te909 committed:
Release 0.101.37

wh1te909 committed:
Release 0.101.38

wh1te909 committed:
Release 0.101.39

wh1te909 committed:
Release 0.101.40

wh1te909 committed:
Release 0.101.43

wh1te909 committed:
Release 0.101.44

wh1te909 committed:
Release 0.101.47

wh1te909 committed:
Release 0.101.48

wh1te909 committed:
Release 0.101.49

wh1te909 committed:
Release 0.101.50

wh1te909 committed:
Release 0.101.50

wh1te909 committed:
Release 0.101.52

wh1te909 committed:
Release 0.101.53

wh1te909 committed:
Release 0.101.54

wh1te909 committed:
Release 0.101.55

wh1te909 committed:
Release 0.101.56

ultimatejedi committed:
Feature: Add dynamic RustDesk URL action button
Adds a new button to the agent summary tab that is conditionally
displayed if a URL Action named 'RustDesk_Connect' exists.

The button's URL is built dynamically by:
- Finding the URL Action 'pattern' from the Vuex store
- Looking up the agent's 'RustDesk_ID' custom field
- Replacing the {{agent.RustDesk_ID}} placeholder
- Leaving the {{global.*}} placeholder intact for the backend to process

NOTE: This requires a new Agent Custom Field labelled 'RustDesk_ID'
      There is also a modified version of the GetRustDeskID script that will populate the custom field.

ultimatejedi commented:
I forgot to add to the notes that this also requires the URL Action to be set up per the documentation in the format of:
rustdesk://connection/new/{{agent.RustDesk_ID}}

This assumes that you have the Agent level Custom Field named "RustDesk_ID".

wh1te909 committed:
Release 0.101.57

wh1te909 committed:
Release 0.101.58

wh1te909 committed:
Release 0.101.59

TeamInfoGit commented:
Hello ! 
I might be dumb (Surely), but i cannot seem to find a way to enable this feature even if my custom fields are correctly named and url action as well ?

Do you maybe have the modified version of GetRustDeskID ? Thanks !

ultimatejedi commented:
Yes, my apologies, I never got alerted that there was a response.

Essentially, Rustdesk can get installed in various directories if you use custom installers or packages.  So this is what I have updated that script to look for.  Essentially, you just need to find out what your install directory and exe's are called.  Hope this helps!

```
$ErrorActionPreference= 'silentlycontinue'

if (Test-Path -Path $env:ProgramFiles\RustDesk\ -PathType Container) {
	cd $env:ProgramFiles\RustDesk\
.\RustDesk.exe --get-id | out-host
} elseif (Test-Path -Path $env:ProgramFiles\RustDesk-MyCustomPath1\ -PathType Container) {
	cd $env:ProgramFiles\RustDesk-MyCustomerPath1\
.\rustdesk-MyCustomApp1.exe --get-id | out-host
} elseif (Test-Path -Path $env:ProgramFiles\RustDesk-MyCustomerPath2\ -PathType Container) {
	cd $env:ProgramFiles\RustDesk-MyCustomerPath2\
.\rustdesk-MyCustomApp2.exe --get-id | out-host
} elseif (Test-Path -Path $env:ProgramFiles\MyCustomerPath3\ -PathType Container) {
	cd $env:ProgramFiles\MyCustomerPath3\
.\MyCustomApp3.exe --get-id | out-host
}
```

ultimatejedi committed:
Security: bump axios 1.12.2 -> 1.16.0 and dompurify 3.2.6 -> 3.4.2

These are the only runtime (browser-shipped) packages with open advisories
in v0.101.59. Bumping eliminates:

  axios:
    - GHSA-43fc-jf86-j433 (DoS via __proto__ in mergeConfig, high)
    - GHSA-3p68-rc4w-qgx5 (NO_PROXY hostname normalization SSRF, moderate)
    - GHSA-fvcv-3m26-pcqx (cloud metadata exfil via header injection, moderate)
    - GHSA-w9j2-pvgh-6h63 (auth bypass via prototype pollution, moderate)
    - GHSA-pmwg-cvhr-8vh7 (NO_PROXY 127.0.0.0/8 bypass, high)
    - GHSA-3w6x-2g7m-8v23 (JSON response tampering, moderate)
    - GHSA-q8qp-cvcw-x6jj (prototype pollution credential injection, high)
  dompurify:
    - GHSA-v8jm-5vwx-cfxm (XSS, moderate)
    - GHSA-v2wj-7wpq-c8vv (XSS, moderate)
    - GHSA-cjmm-f4jc-qw8r (URI validation skip via ADD_ATTR predicate, moderate)

Remaining post-bump vulnerabilities are all in build-tooling
(rollup, esbuild, serialize-javascript, yaml) under @quasar/app-vite
and do not ship to the production dist/. They will be resolved when
upstream tacticalrmm-web migrates to Quasar 2.
